### PR TITLE
Listings

### PR DESCRIPTION
**Pull request by monenavarro**

### CODE REVIEW

File: lib/space.rb [hunk 1]
```diff
@@ -19,22 +20,39 @@ def self.all
...
+    result.map do |space| 
+      Space.new(
+        id: space['id'], 
+        name: space['name'], 
+        price: space['price'], 
+        description: space['description'],
+        user_id: space['user_id']
+     )
+    end
+  end
```
Comment (JordanManu) [inline lib/space.rb:56]: v nice

File: app.rb [hunk 2]
```diff
@@ -22,15 +22,18 @@ class MakersBnB < Sinatra::Base
 
   get '/spaces' do
     @user = User.find(session[:user_id])
-    @spaces = Space.all
+    if @user 
+    @spaces = Space.spaces_listed(session[:user_id])
+    end
```
Comment (almorcrette) [inline app.rb:27]: Current approach is that spaces is not visible if not logged in. That's ok. Justr confirming that this is the approach.

File: spec/features/user_signing_up_spec.rb [hunk 3]
```diff
@@ -14,4 +14,18 @@
...
+    user = User.create(
+        email: 'johndoe@example.com',
+        password: 'Password123'
+      )
+    visit '/users/new'
+    fill_in("email", with: "johndoe@example.com")
+    fill_in('password', with: 'password123')
+    click_on('Sign Up')
+
+    expect(current_path).to eq '/users/new'
```
Comment (JordanManu) [inline spec/features/user_signing_up_spec.rb:28]: great

**Review by JordanManu — COMMENTED** [review]
I think it all looks really good

**Review by almorcrette — APPROVED** [review]
Email already in use implements well.

This implementation means that /spaces is blank unless logged in. And then only displaying ones own spaces.

That's ok, but surely we want the ability for users, or anyone to see all? Is that somewhewre?

Should be implement it on home (/)?

**Pull request merged**